Automattic/wp-calypso
# Login: Cleanup social account creation state and components

#### Changes proposed in this Pull Request

Originally in #14848, we used to use `createSocialUser()` for creating the social account user for signing up with a Google account. However, we no longer do that and `createSocialUser` was just imported but unused at all. So we can delete that one completely.

Furthermore, the `isSocialAccountCreating` selector relies on the `isCreating` social reducer tree property, which is set to true only when `createSocialUser()` is dispatched. So that can no longer occur, and that selector will always be false, therefore it can be removed. 

By removing `isSocialAccountCreating` we also remove the last usage of the `GlobalNotice` component, which even back then was [deemed unnecessary](https://github.com/Automattic/wp-calypso/pull/14848/files#r122887871).

By removing the `GlobalNotice` component, we address one item from #58453.

Finally, this PR is doing some final cleanup by removing all the reducer pieces that will no longer be reachable.

#### Testing instructions

* Verify Calypso builds well and without errors.
* Start as a logged-out user.
* Verify logging in with a Google account still works.
* Verify signing up with a Google account still works.
* Verify regular login still works, too.
* Verify all tests pass.

#### Context

I got here by attempting to test the sole usage of the `GlobalNotice` component in order to refactor it away from `UNSAFE_` React lifecycle methods (see #58453).
